timemachine/cpp/src/nonbonded_all_pairs.hpp

```diff
@@ -46,16 +48,10 @@ template <typename RealType> class NonbondedAllPairs : public Potential {
     unsigned long long *d_gathered_du_dx_;
     unsigned long long *d_gathered_du_dp_;
 
-    // used for hilbert sorting
-    unsigned int *d_bin_to_idx_; // mapping from 256x256x256 grid to hilbert curve index
-    unsigned int *d_sort_keys_in_;
-    unsigned int *d_sort_keys_out_;
-    unsigned int *d_sort_vals_in_;
-    unsigned int *d_sort_storage_;
-    size_t d_sort_storage_bytes_;
-
     cudaEvent_t nblist_flag_sync_event_; // Event to synchronize on
 
+    std::unique_ptr<HilbertCurve> hilbert_sort_;
```

Comment:
nit: could the name of the class be `HilbertSort` too? (seems more descriptive of its purpose vs. just representing an abstract hilbert curve)

Reply:
Seems reasonable to me, had started with that but switched to imitate the python library we use.

Reply:
Renamed in [fbe6ca8](https://github.com/proteneer/timemachine/pull/1113/commits/fbe6ca82469b404246fa0d943e5050e15ec7fcfe)